scanpy/tests/test_highly_variable_genes.py

```diff
@@ -54,6 +57,276 @@ def test_highly_variable_genes_basic():
     assert np.all(np.isin(colnames, hvg_df.columns))
 
 
+def _residual_var_reference(adata, clip=None, theta=100):
+    sc.experimental.pp.normalize_pearson_residuals(adata, clip=clip, theta=theta)
+    residuals = adata.X
+    return np.var(residuals, axis=0)
+
+
+@pytest.mark.parametrize(
+    'sparsity_func', [csr_matrix.toarray, csr_matrix], ids=lambda x: x.__name__
+)
+@pytest.mark.parametrize('dtype', ['float32', 'int64'])
+def test_highly_variable_genes_pearson_residuals_inputchecks(sparsity_func, dtype):
+
+    adata = _prepare_pbmc_testdata(sparsity_func, dtype, small=True)
+
+    # depending on check_values, warnings should be raised for non-integer data
+    if dtype == 'float32':
+
+        adata_noninteger = adata.copy()
+        x, y = np.nonzero(adata_noninteger.X)
+        adata_noninteger.X[x[0], y[0]] = 0.5
+
+        # expecting 0 no-int warnings
+        with pytest.warns(None) as record:
+            sc.experimental.pp.highly_variable_genes(
+                adata_noninteger.copy(),
+                flavor='pearson_residuals',
+                n_top_genes=100,
+                check_values=False,
+            )
+        assert len(record) == 0
+
+        # expecting 1 no-int warning
+        with pytest.warns(None) as record:
+            sc.experimental.pp.highly_variable_genes(
+                adata_noninteger.copy(),
+                flavor='pearson_residuals',
+                n_top_genes=100,
+                check_values=True,
+            )
+        assert len(record) == 1
+        assert "expects raw count data" in record[0].message.args[0]
+
+    # errors should be raised for invalid theta values
+    with pytest.raises(ValueError) as record:
+        sc.experimental.pp.highly_variable_genes(
+            adata.copy(), flavor='pearson_residuals', n_top_genes=100, theta=0
+        )
+    with pytest.raises(ValueError) as record:
+        sc.experimental.pp.highly_variable_genes(
+            adata.copy(), flavor='pearson_residuals', n_top_genes=100, theta=-1
+        )
+
+    # error should be raised for invalid clipping values
+    with pytest.raises(ValueError) as record:
+        sc.experimental.pp.highly_variable_genes(
+            adata.copy(), flavor='pearson_residuals', n_top_genes=100, clip=-1
+        )
+
+
+@pytest.mark.parametrize(
+    'sparsity_func', [csr_matrix.toarray, csr_matrix], ids=lambda x: x.__name__
+)
+@pytest.mark.parametrize('dtype', ['float32', 'int64'])
+@pytest.mark.parametrize('subset', [True, False])
+@pytest.mark.parametrize('clip', [None, np.Inf, 30])
+@pytest.mark.parametrize('theta', [100, np.Inf])
+@pytest.mark.parametrize('n_top_genes', [100, 200])
+def test_highly_variable_genes_pearson_residuals_values(
+    subset, sparsity_func, dtype, clip, theta, n_top_genes
+):
+    adata = _prepare_pbmc_testdata(sparsity_func, dtype, small=True)
+    # cleanup var
+    adata.var.drop(columns=adata.var.columns, inplace=True)
```

Comment:
This is a bit shorter:

```suggestion
    del adata.var
```

Could you replace this all over?

Reply:
fixed!